src/rust/inetstack/protocols/tcp/established/ctrlblk.rs

```diff
@@ -774,20 +784,30 @@ impl<const N: usize> SharedControlBlock<N> {
         // In the normal case, we'll be in either ESTABLISHED or CLOSE_WAIT here (depending upon whether we've received
         // a FIN from our peer yet).  Queue up a FIN to be sent, and attempt to send it immediately (if possible).  We
         // only change state to FIN-WAIT-1 or LAST_ACK after we've actually been able to send the FIN.
-        debug_assert!((self.state == State::Established) || (self.state == State::CloseWait));
-
         // Send a FIN.
+        // TODO: Set state to FIN-WAIT1 if currently established or set to LASTACK if CloseWait.
+        let new_state: State = match self.state {
+            State::Established => State::FinWait1,
+            State::CloseWait => State::LastAck,
+            state => {
+                let cause: String = format!("Cannot close a socket from any other state (current state={:?},", state);
+                warn!("{}", cause);
```

Comment:
Shouldn't this be an error instead?

Reply:
Sure, since it was replacing a debug assert, I thought it should be a warn, but I will change it.